src/App.tsx

```diff
@@ -15,21 +17,57 @@ import { useLocalization, useUser } from 'src/providers';
 import { store } from 'src/redux/store';
 import { getRgbaFromHex } from 'src/utils/color';
 import useDeviceInfo from 'src/utils/useDeviceInfo';
+import useEnvironment from 'src/utils/useEnvironment';
 
 const AcceleratorRouter = React.lazy(() => import('src/scenes/AcceleratorRouter'));
 const TerrawareRouter = React.lazy(() => import('src/scenes/TerrawareRouter'));
 
+// Mixpanel setup
+// Set this to true to enable Mixpanel tracking for the Terraware Dev project
+const enableMixpanelDev = true;
+const { isProduction, isStaging, isDev } = useEnvironment();
+const MIXPANEL_TOKEN = isProduction
```

Comment:
Instead of determining which env var to use here, and exposing all of them, we should provide the correct token for the environment. Then you can just use `process.env.MIXPANEL_TOKEN` instead of determining at runtime.

In the `build` step of the GH workflow you can add something like `REACT_APP_MIXPANEL_TOKEN: ${{ secrets.REACT_APP_MIXPANEL_TOKEN }}`. 

You will also need to add a line in the script that sets the "tier based env var" to the environment before build [here](https://github.com/terraware/terraware-web/blob/main/.github/scripts/set-environment.sh#L28), it should look something like this `REACT_APP_MIXPANEL_TOKEN=${TIER}_REACT_APP_MIXPANEL_TOKEN`. 

The last step is to add the secrets to the repo for `PROD_REACT_APP_MIXPANEL_TOKEN` and `STAGING_REACT_APP_MIXPANEL_TOKEN`.

Reply:
These tokens are not secrets which is why I didn't mind doing it this way (https://developer.mixpanel.com/reference/project-token).  I can still add it to the env vars if you think that would be an improvement but I won't be creating any secrets for these.

Reply:
Env vars are fine. I only mentioned the secrets way because that's currently how other non-secret env vars are passed into the React app. See `REACT_APP_MAPBOX_TOKEN` and `REACT_APP_GOOGLE_KEY`. I'm not sure if there would be more fandanglin' if you are using the proper GH repo env vars.

Either way, I do think it should be passed into the environment instead of determined at runtime.